extension/src/popup/components/manageAssets/ManageAssetRows/index.tsx

```diff
@@ -229,9 +296,26 @@ export const ManageAssetRows = ({
                     isLoading={
                       isActionPending && assetSubmitting === canonicalAsset
                     }
-                    onClick={() => {
-                      if (isBlockedDomain(domain) && !isTrustlineActive) {
-                        handleBlockedDomain({ domain, image, code, issuer });
+                    onClick={async () => {
+                      const resp = await checkForSuspiciousAsset(
+                        code,
+                        issuer,
+                        domain,
+                      );
+                      if (
+                        resp.isInvalidDomain ||
+                        resp.isRevocable ||
+                        resp.isNewAsset
+                      ) {
+                        setShowNewAssetWarning(true);
+                        setNewAssetFlags(resp);
+                        setSuspiciousAssetData({ domain, image, code, issuer });
+                      } else if (
+                        isBlockedDomain(domain) &&
+                        !isTrustlineActive
+                      ) {
+                        setShowBlockedDomainWarning(true);
+                        setSuspiciousAssetData({ domain, image, code, issuer });
                       } else {
                         changeTrustline(code, issuer, !isTrustlineActive);
```

Comment:
it might be nice to move this into a function up above since the logic is getting a little more involved. It's a little nicer to not have bigger pieces of logic mixed in with the JSX

Reply:
👍  https://github.com/stellar/freighter/pull/634/commits/a56954289da8e76c17a1d984d6d54ece5909c2a7